pkg/splunk/test/azureblobclient.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018-2022 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	enterpriseApi "github.com/splunk/splunk-operator/api/v3"
+)
+
+// MockAzureBlobClient is used to store all the objects for an app source
+type MockAzureBlobClient struct {
+	Objects []*MockS3Object
```

Comment:
is this MockS3Object or MockBlobObject?

Reply:
Good point! My plan is revise the test for azureblobclient.go during list api story, So let me take a look in it then if that's ok with you.